scrapper.py

```diff
@@ -20,44 +30,129 @@ class IncorrectNumberOfArticlesError(Exception):
-        pass
+        for url in self.seed_urls:
+            sleep(randint(1, 5))
+            response = requests.get(url)
+            page_bs = BeautifulSoup(response.text, 'lxml')
+            urls = self._extract_url(page_bs)
+            for the_url in urls:
+                if len(self.urls) < self.max_articles:
+                    self.urls.append(the_url)
 
     def get_search_urls(self):
         """
         Returns seed_urls param
         """
-        pass
-
+        return self.seed_urls
+
+class HTMLParser:
+    def __init__(self, article_url, article_id):
+        self.article_url = article_url
+        self.article_id = article_id
+        self.article = Article(article_url, article_id)
+
+    def _fill_article_with_text(self, article_bs):
+        text_bs = article_bs.find('div', class_='b_article-text')
+        self.article.text = text_bs.text
+
+    def _fill_article_with_meta_information(self, article_bs):
+        title_bs = article_bs.find('h1', class_='headline').prettify(formatter='html')
+        self.article.title = title_bs
+
+        author_bs = article_bs.find('div', class_='author')
+        if not author_bs.find('a'):
+            self.article.author = 'NOT FOUND'
+        else:
+            self.article.author = author_bs.find('a').text
+
+        self.article.topics = 'NOT FOUND'
+
+        date_bs = article_bs.find('time').text
+        months = {"января": "01", "февраля": "02", "марта": "03", "апреля": "04",
+                  "мая": "05", "июня": "06", "июля": "07", "августа": "08",
+                  "сентября": "09", "октября": "10", "ноября": "11",
+                  "декабря": "12"}
+        for month in months.keys():
+            if month in date_bs:
+                date_bs = date_bs.replace(month, months[month])
+        self.article.date = datetime.strptime(date_bs, '\n%d %m %Y, %H:%M\n')
+
+    def parse(self):
+        response = requests.get(self.article_url)
+        article_bs = BeautifulSoup(response.text, 'lxml')
+        self._fill_article_with_text(article_bs)
+        self._fill_article_with_meta_information(article_bs)
+        return self.article
 
 def prepare_environment(base_path):
     """
     Creates ASSETS_PATH folder if not created and removes existing folder
     """
-    pass
+    path = Path(base_path)
+    if path.exists():
+        shutil.rmtree(base_path)
+    path.mkdir(exist_ok=True, parents=True)
 
 
 def validate_config(crawler_path):
     """
     Validates given config
     """
-    pass
+    with open(crawler_path, 'r', encoding='utf-8') as file:
+        config = json.load(file)
+    seed_urls = config['seed_urls']
+    max_articles = config['total_articles_to_find_and_parse']
+    if not isinstance(max_articles, int):
+        raise IncorrectNumberOfArticlesError
+    if max_articles <= 0:
+        raise IncorrectNumberOfArticlesError
+    if not isinstance(seed_urls, list) or not seed_urls:
+        raise IncorrectURLError
+    if max_articles > 100:
```

Comment:
take a bigger number, this is too close to how many articles you are required to collect